pybossa/api/__init__.py

```diff
@@ -266,6 +266,9 @@ def _retrieve_new_task(project_id):
     # with the user. So we want to commit that snapshot if this is the first access.
     user_repo.update(user)
 
+    # Allow scheduling a gold-only task if quiz mode is enabled for the user and the project.
+    quiz_mode_enabled = user.get_quiz_in_progress(project) and project.info["quiz"]["enabled"]
```

Comment:
I think instead of using 'in_progress' the right thing should be to use if the user is enable.

Reply:
Thank you. 

_Per discussion: If the project level quiz mode is turned off, and the desired behavior is to have all users have quiz mode off (even if their individual status is set to on ["Must pass"]), then upon disabling quiz mode, programmatically set all users back to disabled quiz state.